lib/allure_report_publisher/lib/providers/gitlab.rb

```diff
@@ -58,28 +58,67 @@ def update_pr_description
         )
       end
 
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
       # Add comment with report url
       #
       # @return [void]
       def add_comment
-        unless comment
+        if main_comment
+          log_debug("Updating summary in comment with id #{discussion.id} in mr !#{mr_iid}")
+
+          client.edit_merge_request_note(project, mr_iid, main_comment.id,
+                                         url_section_builder.comment_body(main_comment.body))
```

Comment:
Minor nitpick that should probably be enforced by rubocop, all methods in the project use the multiline formatting with braces on separate lines, I would like to keep that consistent:

```suggestion
          client.edit_merge_request_note(
            project, 
            mr_iid, 
            main_comment.id,
            url_section_builder.comment_body(main_comment.body)
          )
```